Sources/NIOHTTPClient/HTTPHandler.swift

```diff
@@ -376,12 +378,19 @@ internal class TaskHandler<T: HTTPClientResponseDelegate>: ChannelInboundHandler
                 break
             default:
                 self.state = .body
-                let future = self.delegate.didReceivePart(task: self.task, eventLoop: context.eventLoop, body)
+                let future = self.delegate.didReceivePart(task: self.task, body)
                 self.mayRead = false
-                future.whenComplete { _ in
-                    self.mayRead = true
-                    if self.pendingRead {
-                        context.read()
+                future.whenComplete { result in
+                    switch result {
+                    case .success:
+                        self.mayRead = true
+                        if self.pendingRead {
+                            context.read()
+                        }
+                    case .failure(let error):
+                        self.state = .end
+                        self.delegate.didReceiveError(task: self.task, error)
```

Comment:
Hmm, would the user expect to receive this callback when they triggered the error themselves?

Reply:
What if they use `NonBlockingIO.readChunked`, for example, and just pass returned future as the return from the delegate?

Reply:
Ah good point - in that case they won't know that the read failed, so `didReceiveError()` is their only hope. Thanks :)